Frameworks/Foundation/NSURLRequest.mm

```diff
@@ -138,6 +140,8 @@ - (void)encodeWithCoder:(NSCoder*)coder {
     [coder encodeBool:_HTTPShouldUsePipelining forKey:@"HTTPShouldUsePipelining"];
     [coder encodeBool:_HTTPShouldHandleCookies forKey:@"HTTPShouldHandleCookies"];
     [coder encodeBool:_allowsCellularAccess forKey:@"allowsCellularAccess"];
+
+    [coder encodeObject:_properties forKey:@"_properties"];
```

Comment:
In fact, it's actively bad to do that: it makes it so that all requests that get decoded get a properties dictionary even if they didn't have properties on the other end.

Use objective-c's nil propagation to your advantage!